apiserver/facades/client/modelupgrader/controller_upgrader_test.go

```diff
@@ -488,13 +513,54 @@ func (u *controllerUpgraderAPISuite) TestUpgradeModelMapErrAgentStreamNotValidTo
 	c.Assert(res, tc.DeepEquals, params.UpgradeModelResult{})
 }
 
+// TestUpgradeModelMapOtherErrorsToServerError tests that the
+// [controllerupgradererrors.DowngradeNotSupported] is mapped to a not
+// supported error. This is a sad case.
+func (u *controllerUpgraderAPISuite) TestUpgradeModelMapErrDowngradeNotSupportedToNotSupported(c *tc.C) {
+	ctrl := u.setup(c)
+	defer ctrl.Finish()
+
+	u.authorizer.EXPECT().AuthClient().Return(true)
+	u.authorizer.EXPECT().HasPermission(
+		gomock.Any(),
+		permission.SuperuserAccess,
+		u.controllerTag,
+	).Return(nil)
+	u.check.EXPECT().ChangeAllowed(gomock.Any()).Return(nil)
+	u.upgraderService.EXPECT().UpgradeController(gomock.Any()).Return(
+		semversion.Zero,
```

Comment:
Would you mind changing the version we use for this test? A zero value version should be considered invalid and user supplied error. When I read the test I want to understand the contract and behavior.